Add 'method' parameter to forgiving_round method

Mofeywalker opened the pull request:
Fixes #21707

## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

Mofeywalker committed:
Add 'method' parameter to forgiving_round method

Fixes #21707

homeassistant commented:
Hi @Mofeywalker, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#21708).

Once you do that we will be able to review and accept this pull request.

Thanks!

Danielhiversen commented:
Could you add a test https://github.com/home-assistant/home-assistant/blob/935e5c67a3e9892f9bc3611c8ee4efa629bedea4/tests/helpers/test_template.py#L85

Mofeywalker committed:
fix rounding behavior in round() filter

Mofeywalker committed:
add test cases for new rounding behaviour

Mofeywalker commented:
> Could you add a test
> 
> [home-assistant/tests/helpers/test_template.py](https://github.com/home-assistant/home-assistant/blob/935e5c67a3e9892f9bc3611c8ee4efa629bedea4/tests/helpers/test_template.py#L85)
> 
> Line 85 in [935e5c6](/home-assistant/home-assistant/commit/935e5c67a3e9892f9bc3611c8ee4efa629bedea4)
> 
>  def test_rounding_value(self):

I checked again and found that the python builtin round()  is used. I added code which copies jinja2's rounding behavior and added tests

OttoWinter commented:
I believe these should all be their own methods, like `10.5 | ceil` etc. No programming language I know uses an optional parameter to specify if ceil/floor should be used.

Mofeywalker commented:
I can implement these as own methods.
I implemented the same behavior as [Jinja](http://jinja.pocoo.org/docs/2.10/templates/#list-of-builtin-filters)

OttoWinter commented:
Ok if jinja does it that way then that's the preferred way here 👍
